### PR TITLE
Add TAO (new formula)

### PR DESCRIPTION
**Pull request by jwillemsen**
- [ ] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [ ] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

**Comment by jwillemsen** [comment]
Based on the ace formula

File: Formula/tao.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  homepage "https://www.dre.vanderbilt.edu/~schmidt/TAO.html"
+  url "http://download.dre.vanderbilt.edu/previous_versions/ACE+TAO-6.4.0.tar.bz2"
+  sha256 "8177d4f16bf013f7a003a7d1f83de9fda646406754c1f4cc0b0964a4b839eb0f"
+
+  bottle do
+    cellar :any
+    sha256 "0b53c1e14820be00af7115e9afefa09a380803589597038d962bb8ab16b79331" => :el_capitan
+    sha256 "d3e7e45197f7d7643d675ff84057df769947f6c0a9185388d23b2a28375760f0" => :yosemite
+    sha256 "72f21357797c350fe547a7978f402e564cf328c3662a0e321d1ec293f4899139" => :mavericks
+  end
```
Comment (MikeMcQuaid) [inline Formula/tao.rb]: Remove this block; we'll add then when we pull.

File: Formula/tao.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                   "DESTDIR=",
+                   "INST_DIR=/tao",
+                   "debug=0",
+                   "shared_libs=1",
+                   "static_libs=0",
+                   "install"
+
+  end
+
+  test do
```
Comment (MikeMcQuaid) [inline Formula/tao.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Formula/tao.rb [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    name = MacOS.cat.to_s.delete "_"
+    ln_sf "config-macosx-#{name}.h", "ace/config.h"
+    ln_sf "platform_macosx_#{name}.GNU", "include/makeinclude/platform_macros.GNU"
+
+    # Set up the environment the way TAO expects during build.
+    ENV["ACE_ROOT"] = buildpath
+    ENV["TAO_ROOT"] = "#{buildpath}/TAO"
+    ENV["DYLD_LIBRARY_PATH"] = "#{buildpath}/lib"
+
+    # Done! We go ahead and build.
```
Comment (MikeMcQuaid) [inline Formula/tao.rb]: Can remove all these comments.

**Pull request closed without merging**